rfcs/12-primitive-shapes.md

```diff
@@ -0,0 +1,538 @@
+  translation: Vec2,
+}
+impl Meshable for CircleCollider2d {}
+impl Collider2d for CircleCollider2d {}
+
+// Box Types
+
+struct Box2d {
+  half_extents: Vec2,
+}
+impl Meshable for Box2d
+
+struct BoundingBox2d {
+  box: Box2d,
+  translation: Vec2,
+}
+impl Meshable for BoundingBox2d {}
+impl Bounding2d for BoundingBox2d {}
+type Aabb2d = BoundingBox2d;
+
+struct BoxCollider2d {
+  box: Box2d,
+  translation: Vec2,
+  rotation: Mat2,
+}
+impl Meshable for BoxCollider2d {}
+impl Collider2d for BoxCollider2d {}
+type Obb2d = BoxCollider2d;
+
+// Capsule Types
+
+struct Capsule2d {
+  // Height of the cylindrical section
+  height: f32,
+  radius: f32,
+}
+impl Meshable for Capsule2d {}
+
+struct CapsuleCollider2d {
+  capsule: Capsule2d,
+  translation: Vec2,
+  rotation: Mat2,
+}
+impl Meshable for CapsuleCollider2d {}
+impl Collider2d for CapsuleCollider2d {}
+
+```
+### Lack of `Transform`s
+
+Primitives colliders and bounding volumes are fully defined in space, and do not use `Transform` or `GlobalTransform`. This is an intentional decision. Because transforms can include nonuniform scale, they are fundamentally incompatible with shape primitives. We could use transforms in the future if the `translation`, `rotation`, and `scale` fields were distinct components, and shape primitives could be bundled with a `translation` or `rotation` if applicable.
+
+#### Cache Efficiency
+
+- Some primitives such as AABB and Sphere don't need a rotation (or scale) to be fully defined. 
+- Using a `GlobalTransform` adds an unused Quat and Vec3 to the cache line.
+- This is especially important for AABBs and Spheres, because they are fundamental to broad phase collision detection and BV(H), and as such need to be as efficient as possible.
+- This also applies to all other primitives, which don't use the `scale` field of the `GlobalTransform`.
+
+#### Ergonomics
+
+- Primitives need to be fully defined in world space to compute collision or bounding.
+- By making the primitive components fully defined and standalone, computing operations is as simple as: `primitive1.some_function(primitive_2)`, instead of also having query and pass in 2 `GlobalTransform`s in the (hopefully) correct order.
+
+#### Use with Transforms
+
+- The meshes generated from these primitives can be transformed freely. The primitive types, however, do not interact with transforms for the reasons stated above.
+
+### Bounding Boxes/Volumes
+
+Because bounding volumes and colliders are fully defined in world space, this leads to the natural question of how they are kept in sync with their parent. An implementation should provide a system similar to transform propagation, that would update the primitive as well as its translation and rotation if applicable. Further details are more appropriate for a subsequent bounding RFC/PR. The important point to consider is how this proposal provides common types that can be used for this purpose, whether for internal or external crates.
```

Comment:
I think the answer to this question is important for the choice to have fully defined volumes and colliders. It doesn't need implementation, but some things like system ordering become very relevant.

Reply:
I'd counter this by asking what the alternative to fully defining them is? I think I've outlined reasonably well through this doc why using `Transform`s isn't an acceptable solution, at least until we can compose transforms from independent components.

I think if we wait on answering this question, we'll never get primitives out the door - we have to draw the line somewhere. If the scope of this RFC is appropriate, then at some point we need to make the call if these primitives are good enough and well justified. I only say this because I've done enough prototyping with bounding volumes in Bevy to know that it is a very deep rabbit hole, and I don't think answering that question in any more depth is possible without a full bounding proposal.

Reply:
I don't mean holding off on adding Primitives, sorry. I think just discussing it in a short paragraph would be good.

Reply:
I've added a note - thanks!